packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextPrimitivesConversions.h

```diff
@@ -94,6 +95,43 @@ inline static NSUnderlineStyle RCTNSUnderlineStyleFromTextDecorationStyle(
   }
 }
 
+inline UIColor *_Nullable RCTUIColorFromDynamicColor(const facebook::react::Color &color)
+{
+  auto dynamicColor = color.getDynamicColor();
+  int32_t light = dynamicColor.lightColor;
+  int32_t dark = dynamicColor.darkColor;
+  int32_t highContrastLight = dynamicColor.highContrastLightColor;
+  int32_t highContrastDark = dynamicColor.highContrastDarkColor;
+
+  UIColor *lightColor = [RCTConvert UIColor:@(light)];
+  UIColor *darkColor = [RCTConvert UIColor:@(dark)];
+  UIColor *highContrastLightColor = [RCTConvert UIColor:@(highContrastLight)];
+  UIColor *highContrastDarkColor = [RCTConvert UIColor:@(highContrastDark)];
+
+  if (lightColor != nil && darkColor != nil) {
+    UIColor *color = [UIColor colorWithDynamicProvider:^UIColor *_Nonnull(UITraitCollection *_Nonnull collection) {
+      if (collection.userInterfaceStyle == UIUserInterfaceStyleDark) {
+        if (collection.accessibilityContrast == UIAccessibilityContrastHigh && highContrastDarkColor != nil) {
+          return highContrastDarkColor;
+        } else {
+          return darkColor;
+        }
+      } else {
+        if (collection.accessibilityContrast == UIAccessibilityContrastHigh && highContrastLightColor != nil) {
+          return highContrastLightColor;
+        } else {
+          return lightColor;
+        }
+      }
+    }];
+    return color;
+  } else {
+    return nil;
+  }
+
+  return nil;
+}
```

Comment:
This is duplicated code.
We should fix it by create a separate pod for `React-TextLayoutManagerApple` that can depends on `React-Core` and access the `RCTConversion.h` file.

Could you add a `TODO` to do this, please? 

**Edit:** Probably, this all file require a deeper inspection. I suspect the whole file have duplicated code. 😭